spec/solver.spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rspec/autorun'
+require_relative '../solver'
+
+RSpec.describe ' Solver' do
+  describe '#factorial' do
+    it 'should return factorial of a number' do
+      res = Solver.new
+      expect(res).to receive(:factorial).with(5).and_return(120)
+      expect(res.factorial(5)).to eq(120)
+    end
+  end
```

Comment:
- [ ] Kindly include a scenario for when the number is negative, in which case it should raise an exception.

- [ ] Include test for when the number is 0, the result should be 1

Reply:
Ok, let us fix that. Thank you